internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -152,42 +152,39 @@ func resourceRecoveryServicesVaultCreate(d *pluginsdk.ResourceData, meta interfa
 
 	log.Printf("[DEBUG] Creating Recovery Service %s", id.String())
 
-	existing, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	existing, err := client.Get(ctx, id)
 	if err != nil {
-		if !utils.ResponseWasNotFound(existing.Response) {
+		if !response.WasNotFound(existing.HttpResponse) {
 			return fmt.Errorf("checking for presence of existing Recovery Service %s: %+v", id.String(), err)
 		}
 	}
-	if existing.ID != nil && *existing.ID != "" {
-		return tf.ImportAsExistsError("azurerm_recovery_services_vault", *existing.ID)
+	if existing.Model != nil && existing.Model.Id != nil && *existing.Model.Id != "" {
+		return tf.ImportAsExistsError("azurerm_recovery_services_vault", *existing.Model.Id)
```

Comment:
this can become:

```suggestion
	if !response.WasNotFound(existing.HttpResponse) {
		return tf.ImportAsExistsError("azurerm_recovery_services_vault", id.ID())
```